Applications/Spire/Include/Spire/Ui/DurationBox.hpp

```diff
@@ -92,6 +92,24 @@ namespace Styles {
       void on_submit();
       void on_reject();
   };
+
+  /**
+   * Returns a newly constructed time box.
+   * @param time The time displayed in the time box.
+   * @param parent The parent widget.
+   */
+  DurationBox* make_time_box(
+    const boost::optional<boost::posix_time::time_duration>& time,
+    QWidget* parent = nullptr);
+
+  /**
+   * Returns a newly constructed time box without a default time.
+   * @param parent The parent widget.
+   */
+  DurationBox* make_time_box(QWidget* parent = nullptr);
+
+  /** Returns a model which represents a time of day. */
+  std::shared_ptr<LocalOptionalDurationModel> make_time_of_day_model();
```

Comment:
Don't expose implementation details, it makes modifying the API harder in the future, return an ```OptionalDurationModel```.